types/testingutils/beacon_node.go

```diff
@@ -486,56 +503,84 @@ func (bn *TestingBeaconNode) GetBeaconNetwork() types.BeaconNetwork {
 }
 
 // GetAttestationData returns attestation data by the given slot and committee index
-func (bn *TestingBeaconNode) GetAttestationData(slot spec.Slot, committeeIndex spec.CommitteeIndex) (ssz.Marshaler, spec2.DataVersion, error) {
-	return TestingAttestationData, spec2.DataVersionPhase0, nil
+func (bn *TestingBeaconNode) GetAttestationData(slot phase0.Slot, committeeIndex phase0.CommitteeIndex) (ssz.Marshaler, spec.DataVersion, error) {
+	return TestingAttestationData, spec.DataVersionPhase0, nil
 }
 
 // SubmitAttestation submit the attestation to the node
-func (bn *TestingBeaconNode) SubmitAttestation(attestation *spec.Attestation) error {
+func (bn *TestingBeaconNode) SubmitAttestation(attestation *phase0.Attestation) error {
 	r, _ := attestation.HashTreeRoot()
 	bn.BroadcastedRoots = append(bn.BroadcastedRoots, r)
 	return nil
 }
 
 // GetBeaconBlock returns beacon block by the given slot and committee index
-func (bn *TestingBeaconNode) GetBeaconBlock(slot spec.Slot, committeeIndex spec.CommitteeIndex, graffiti, randao []byte) (ssz.Marshaler, spec2.DataVersion, error) {
-	return TestingBeaconBlock, spec2.DataVersionBellatrix, nil
+func (bn *TestingBeaconNode) GetBeaconBlock(slot phase0.Slot, committeeIndex phase0.CommitteeIndex, graffiti, randao []byte) (ssz.Marshaler, spec.DataVersion, error) {
+	return TestingBeaconBlock, spec.DataVersionCapella, nil
 }
 
 // SubmitBeaconBlock submit the block to the node
-func (bn *TestingBeaconNode) SubmitBeaconBlock(block *bellatrix.SignedBeaconBlock) error {
-	r, _ := block.HashTreeRoot()
+func (bn *TestingBeaconNode) SubmitBeaconBlock(block *spec.VersionedBeaconBlock, sig phase0.BLSSignature) error {
+	var r [32]byte
+	switch block.Version {
+	case spec.DataVersionCapella:
+		if block.Capella == nil {
```

Comment:
should we also check if the sig is nil?
not sure

Reply:
not needed will fail validation